### PR TITLE
Add new GP Connect integration types

### PR DESCRIPTION
**Pull request by bjss-colin**
Since we have an upcoming set of tickets in which we'll be moving Integrations from code to the database, I have made this change following the current approach to handling integrations. This includes a migration script that'll migrate the data within the Solutions JSON column as well as the Competition integration qualifiers.

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/InteropGpConnectIntegrationType.cs [hunk 1]
```diff
@@ -4,10 +4,14 @@ namespace NHSD.GPIT.BuyingCatalogue.EntityFramework.Catalogue.Configuration;
 
 public enum InteropGpConnectIntegrationType
 {
-    [Display(Name = "GP Connect - HTML View")]
-    HTML_View,
-    [Display(Name = "GP Connect - Appointment Booking")]
-    Appointment_Booking,
-    [Display(Name = "GP Connect - Structured Record")]
-    Structured_Record,
+    [Display(Name = "GP Connect - Access Record HTML")]
```
Comment (Patrick-M-BJSS) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/InteropGpConnectIntegrationType.cs]: Do these names want dashes in?

File: src/NHSD.GPIT.BuyingCatalogue.Framework/Constants/Interoperability.cs [hunk 2]
```diff
@@ -26,9 +26,11 @@ public static class Interoperability
 
         public static Dictionary<string, string> GpConnectIntegrations => new()
         {
-            { "HTML View", "GP Connect - HTML View" },
-            { "Appointment Booking", "GP Connect - Appointment Booking" },
-            { "Structured Record", "GP Connect - Structured Record" },
+            { "Access Record HTML", "GP Connect - Access Record HTML" },
```
Comment (Patrick-M-BJSS) [inline src/NHSD.GPIT.BuyingCatalogue.Framework/Constants/Interoperability.cs]: Above comment about dashes

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/InteropGpConnectIntegrationType.cs (hunk 1 above)
Comment (ghost) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/InteropGpConnectIntegrationType.cs]: Done, good spot 👍

File: src/NHSD.GPIT.BuyingCatalogue.Framework/Constants/Interoperability.cs (hunk 2 above)
Comment (ghost) [reply src/NHSD.GPIT.BuyingCatalogue.Framework/Constants/Interoperability.cs]: Resolved

**Review by Patrick-M-BJSS — APPROVED** [review]

**Review by imranbashir01 — APPROVED** [review]

**Pull request merged**